pkg/mod/module.go

```diff
@@ -53,3 +53,17 @@ func Find(name string) (*Module, error) {
 	}
 	return mod, nil
 }
+
+func hasPathPrefix(prefix, s string) bool {
+	switch {
+	default:
+		return false
+	case len(s) == len(prefix):
+		return s == prefix
+	case len(s) > len(prefix):
+		if prefix != "" && prefix[len(prefix)-1] == filepath.Separator {
```

Comment:
This part seems strange, `filepath.Clean` would remove `filepath.Separator` at the end